ocaml/xapi/xapi_pusb.ml

```diff
@@ -31,6 +32,55 @@ let create ~__context ~uSB_group ~host ~other_config ~path
   debug "PUSB ref='%s' created" (Ref.string_of pusb);
   pusb
 
+let scan_start ~__context =
+  let open USB in
+  let host = Helpers.get_localhost ~__context in
+  let usbs = get_usbs ~__context in
+  let existing_pusbs = List.filter (fun (rf, rc) -> rc.API.pUSB_host = host) (Db.PUSB.get_all_records ~__context) in
+  let existing_usb_set = get_exsiting_usb_set existing_pusbs in
+  let current_usb_set = get_current_usb_set usbs in
+  (* Get union of two usb sets *)
+  let usb_union = USBSet.union existing_usb_set current_usb_set in
+  (* Get the usb set to add and delete *)
+  let usb_to_del = USBSet.diff usb_union current_usb_set in
+  let usb_to_add = USBSet.diff usb_union existing_usb_set in
+
+  (* Create the newly added pusbs *)
+  USBSet.iter (fun s -> let self = create ~__context ~uSB_group:(Ref.null) ~host ~other_config:[] ~path:s.path ~vendor_id:s.vendor_id
```

Comment:
```
USB.USBSet.iter (....) (USB.USBSet.diff local_usb known_usb)
```
This should create all devices that are local but not yet known by xapi.

Reply:
updated